chips/apollo3/src/gpio.rs

```diff
@@ -8,13 +8,72 @@ use kernel::common::registers::{register_bitfields, register_structs, ReadWrite}
 use kernel::common::StaticRef;
 use kernel::hil::gpio;
 
-const GPIO_BASE: StaticRef<GpioRegisters> =
+pub const GPIO_BASE: StaticRef<GpioRegisters> =
```

Comment:
I think that base address, together with the register definitions should probably be split out in a separate `.rs`, as they remain the same regardless what driver we use. Allows to avoid reinventing the wheel for the "board" drivers.

Basically decouple implementation from the register and base definitions.